parser/parser/cc/driver.cc

```diff
@@ -14,7 +14,9 @@ typedruby27::typedruby27(std::string_view source, const struct builder &builder)
 
 ForeignPtr typedruby27::parse(SelfPtr self, bool trace) {
     bison::typedruby27::parser p(*this, self);
+#if DEBUG_MODE
     p.set_debug_level(trace ? 1 : 0);
+#endif
```

Comment:
What do you think about including `sorbet_version.h` here and using `::sorbet::debug_mode`?

This would be the first place outside of `sorbet_version.{h,c}` that we use the macro variable directly.

Reply:
Using `::sorbet::debug_mode` doesn't work, because `set_debug_level` doesn't exist on `p` in non-debug builds.  (`if constexpr` doesn't save you either, because the behavior that you would want here isn't available outside of templates.)

The other alternative is to compile the grammar once with tracing and once without, then switch between them depending on the trace level.  This requires some build hackery and duplicated code, but at least would make the fast path faster and the test could remain as-is.